// WARNING: This output format is intended for human consumers only
// and is subject to change without notice. Knock yourself out.
fn main() -> () {
    let mut _0: ();
    let _1: i32;
    let mut _2: bool;
    let mut _3: bool;
    let mut _4: !;
    scope 1 {
        debug num => _1;
    }

    bb0: {
        _1 = Add(const 20_i32, const 22_i32);
        _3 = Eq(_1, const 42_i32);
        _2 = Not(move _3);
        switchInt(move _2) -> [0: bb2, otherwise: bb1];
    }

    bb1: {
        _4 = core::panicking::panic(const "assertion failed: num == 42");
    }

    bb2: {
        return;
    }
}
